testing/sdk_auth.py

```diff
@@ -366,8 +376,8 @@ def __encode_secret(self, keytab_path: str) -> List[str]:
                     keytab = f.read()
 
                 base64_encoding = base64.b64encode(keytab).decode("utf-8")
-                with open(base64_encoded_keytab_path, "w") as f:
-                    f.write(base64_encoding)
+                with open(base64_encoded_keytab_path, "wb") as f:
```

Comment:
Without this we get:

```
testing/sdk_auth.py:379: error: Incompatible types in assignment (expression has type "TextIO", variable has type "BinaryIO")
testing/sdk_auth.py:380: error: No overload variant of "write" of "BinaryIO" matches argument type "str"
testing/sdk_auth.py:380: note: Possible overload variants:
testing/sdk_auth.py:380: note:     def write(self, s: bytearray) -> int
testing/sdk_auth.py:380: note:     def write(self, s: bytes) -> int
```

Reply:
Hm, `decode` returns a `str`, so passing `base64_encoding` to `open(..., "w").write()` should just work, no?

I think all you need to do here is use a variable name different than `f`, and not change to binary mode. In theory this should be all ASCII, but safe is better than sorry.